libs/wingc/src/lsp/rename_visitor.rs

```diff
@@ -29,17 +29,17 @@ impl<'a> RenameVisitor<'a> {
 		self
 			.linked_symbols
 			.iter()
-			.any(|s: &LinkedSymbol<'a>| symbol.same(&s.symbol))
+			.any(|s: &LinkedSymbol<'a>| symbol.same(&s.symbol) || s.references.iter().any(|r| symbol.same(r)))
 	}
 
-	fn add_reference_symbol(&mut self, symbol: &'a Symbol) {
+	fn add_reference_symbol(&mut self, symbol: &'a Symbol, symbol_env: Option<&UnsafeRef<SymbolEnv>>) {
 		// symbols that appear in let/if lef statements will point to a prev declaration of a variable of the same name if exists
 		// this is why we add them in advance during visit_statement
 		// the other condition is for "this" that points to the "new" keyword for some reason
-		if self.is_symbol_linked(symbol) {
+		if self.is_symbol_linked(symbol) || symbol.name == "this" {
```

Comment:
Does this need to handle "super" as well?

Reply:
super seems to be handled already (probably never included)